File: Unity-AVL/Assets/Scripts/Terrain/SumoJunction.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class SumoJunction : MonoBehaviour
{
    protected string junctionId = null;
    protected List<Vector3> shape = new List<Vector3>();

    public void Init(JunctionInitData initData) {
        this.gameObject.name += "_" + initData.junctionId;

        this.junctionId = initData.junctionId;
        this.transform.position = new Vector3(
            (float)initData.position[0],
            0,
            (float)initData.position[1]
        );

        foreach (List<double> vertexPos in initData.shape) {
            Vector3 vertex = new Vector3(
                (float)vertexPos[0],
                0,
                (float)vertexPos[1]
            );

            this.shape.Add(vertex);
        }

    }
}

File: Python-AVL_Deprecated/SumoJunction.py
import json

class SumoJunction:
    def __init__(self, ID, position, shape):
        self.ID = ID
        self.position = position
        self.shape = shape

    def jsonInitData(self):
        data = {}

        data["junctionId"] = self.ID
        data["position"] = self.position
        data["shape"] = self.shape

        return json.dumps(data)